File: app/src/main/java/com/jiujiu/mvvmTemplate/di/scope/WorkerKey.java
package com.jiujiu.mvvmTemplate.di.scope;

import androidx.work.ListenableWorker;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import dagger.MapKey;

@Retention(RetentionPolicy.RUNTIME)
@MapKey
public @interface WorkerKey {
    Class<? extends ListenableWorker> value();
}

File: app/src/main/java/com/jiujiu/mvvmTemplate/di/scope/WorkerKey.kt
package com.jiujiu.mvvmTemplate.di.scope

import androidx.work.ListenableWorker
import dagger.MapKey
import kotlin.reflect.KClass

@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@MapKey
annotation class WorkerKey(val value: KClass<out ListenableWorker>)
